fe-library-app/src/components/AddAuthor/AddAuthor.tsx

```diff
@@ -0,0 +1,90 @@
+import { FormEvent, useState } from 'react'
+
+import { createAuthor } from '../../services/AuthorService'
+import styles from './AddAuthor.module.css'
+
+interface AddAuthorProps {
+  onFinish: () => void,
+  onHide?: () => void
+}
+
+const AddAuthor = (props: AddAuthorProps) => {
+  const [ authorFirstNameError, setAuthorFirstNameError ] = useState(false)
+  const [ authorLastNameError, setAuthorLastNameError ] = useState(false)
+  const [ authorFirstName, setAuthorFirstName ] = useState('')
+  const [ authorLastName, setAuthorLastName ] = useState('')
+
+  const handleAuthorFirstNameBlur = () => {
+    if (authorFirstName.trim().length === 0) {
+      setAuthorFirstNameError(true)
+    }
+  }
+
+  const handleAuthorLastNameBlur = () => {
+    if (authorLastName.trim().length === 0) {
+      setAuthorLastNameError(true)
+    }
+  }
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    event.stopPropagation()
+    if (authorFirstNameError || authorLastNameError) {
+      alert('Invalid requested inputs!')
+      return
+    }
+
+    createAuthor({
+      FirstName: authorFirstName,
+      LastName: authorLastName
+    }).then(() => {
+      props.onFinish()
```

Comment:
You should use more descriptive name to know exactly what is onFinish used for, something like onCreateAuthorSuccess - finish can be both for success and error, it only says that it's done

Reply:
Understand